src/GrammarReader.hs

```diff
@@ -0,0 +1,199 @@
+-- relation without negative or positive formula (with only one word) is the relation of context-free grammar.
+pVeryLongRule :: Parser [Symbol]
+pVeryLongRule = do
+    symbols <- pWord
+    symbols <- (++) <$> pure symbols <*> concat <$> many pPositiveConjunction
+    void ("_")
+    symbols <- (++) <$> pure symbols <*> concat <$> many pNegativeConjunction
+    return symbols
+
+pNegativeConjunction :: Parser [Symbol]
+pNegativeConjunction = do
+    symbols1 <- pConjunction
+    symbols2 <- pNegation
+    symbols3 <- pWord
+    symbols <- pure ((++) (symbols1 : symbols2 : []) symbols3)
+    return symbols
+
+pPositiveConjunction :: Parser [Symbol]
+pPositiveConjunction = do
+    symbols <- pConjunction
+    symbols <- (++) <$> pure (symbols : []) <*> pWord
+    return symbols
+
+pPositiveFormula :: Parser [Symbol]
+pPositiveFormula = do
+    symbols <- pWord
+    symbols <- ((++) <$> pure symbols <*> concat <$> many pPositiveConjunction)
+    return symbols
+
+pNegativeFormula :: Parser [Symbol]
+pNegativeFormula = do
+    symbols <- pNegation
+    symbols <- (++) <$> pure (symbols : []) <*> pWord
+    symbols <- ((++) <$> pure symbols <*> concat <$> many pNegativeConjunction)
+    return symbols
+
+-- |Parsers for set of terminals and nonterminals.
+pNonterminals :: Parser (Set Nonterminal)
+pNonterminals = do
+    nonterminals <- Set.fromList <$> ((++) <$> many (void " " *> pNonterminal) <*> (pure []))
+    return nonterminals
+
+pTerminals :: Parser (Set Terminal)
+pTerminals = do
+    terminals <- Set.fromList <$> ((++) <$> many (void " " *> pTerminal) <*> (pure []))
+    return terminals
+
+-- |Parser for whole Grammar.
+-- |Order of grammar components in input is:
+-- |{Start symbol};{Set of nonterminals};{Set of terminals};{Relations}
+pGrammar :: Parser Grammar
+pGrammar = do
+    startSymbol <- pNonterminal
+    void (";")
+    nonterminals <- pNonterminals
+    void (";")
+    terminals <- pTerminals
+    relations <- pRelations
+    grammar <- Grammar <$> pure (nonterminals, terminals, relations, startSymbol)
+    return grammar
+
+parseFromFile p errorFile grammarFile = runParser p errorFile <$> ((fromString) <$> readFile grammarFile)
+
+-- |Method for classifying type of input grammar.
+checkGrammarType :: Grammar -> GrammarType
+checkGrammarType (Grammar (_, _, setOfRelations, _)) =
+    checkGrammarType' $ (concat . map (\ (Relation (_, symbols)) -> symbols) . Set.toList) setOfRelations
+
+checkGrammarType' :: [Symbol] -> GrammarType
+checkGrammarType' symbols
+    | (List.elem (O Conjunction) symbols) && (List.elem (O Negation) symbols) = Boolean
+    | (List.elem (O Conjunction) symbols) && not (List.elem (O Negation) symbols) = Conjunctive
+    | not (List.elem (O Conjunction) symbols) && not (List.elem (O Negation) symbols) = CFG
+
+
+--temporary added deriving show to all types in GrammarType module
+main = do
+    putStrLn "Enter the name of file with grammar. Grammar file should be in working directory."
```

Comment:
Just use https://hackage.haskell.org/package/parseargs or a similar package to create a clear command-line interface.